radareorg/r2ghidra-dec
# Annotators for new annotations

**Detailed description**
As part of the [Improving Decompiler Widget](https://github.com/orgs/radareorg/projects/1) project, many new types of annotations have been added. This was merged to radare2 by [PR #17429](https://github.com/radareorg/radare2/pull/17429).

The annotators for these new annotations were implemented and merged to the `decompiler-refactoring` branch by the following PRs:
1. #123 - Function name annotation annotator implemented.
2. #124 - Annotator for constant and global variables.
3. #128 - Annotator for local variables and function parameters.


...

**Test plan**
1. Fetch the master of radare2 and compile.
2. Compile this PR.
3. Run tests in `r2ghidra-dec` and make sure the new tests are passing. New tests added tests all the annotators that were implemented by the linked PRs.
4. Run `pdgj~{}` on some small executables to see if the output is as expected.
5. Have a look at the code.
6. If you want, you can fetch the `decompiler-refactoring` branch in Cutter and test all the new actions in the decompiler context menu. But I don't think that is necessary.
